src/structure.cpp

```diff
@@ -5960,15 +5960,14 @@ void hqReward(UBYTE losingPlayer, UBYTE rewardPlayer)
 
 // Return true if structure is a factory of any type.
 //
-bool StructIsFactory(const STRUCTURE *Struct)
+bool STRUCTURE::isFactory() const
 {
```

Comment:
Just a note: Anywhere that `StructIsFactory` was previously called, we should ensure there’s a null check, since we’re removing the one that was here.

Reply:
Good point. I'll check all call sites and add more checks if needed.

Reply:
Added the checks wherever it made sense. In most places it's either already checked prior or `psStruct` is already used above, so that one will surely catch a SIGSEGV even before getting to the `isFactory()` call.